sound/soc/sof/Kconfig

```diff
@@ -252,6 +252,13 @@ config SND_SOC_SOF_PROBE_WORK_QUEUE
 	  When selected, the probe is handled in two steps, for example to
 	  avoid lockdeps if request_module is used in the probe.
 
+# Supported IPC versions
+config SND_SOC_SOF_IPC3
+	bool
+
+config SND_SOC_SOF_INTEL_IPC4
```

Comment:
@ujfalusi  Not sure if it is better to define as SND_SOC_SOF_INTEL_IPC4 for future usage on other platforms(e.g. imx/amd/mediatek...)? Thanks.

Reply:
@yaochunhung, it is not that simple for other platforms to support this new IPC version. We have tried to make the high level code for IPC4 as platform independent as it can be, but you would need to have implementation in firmware and in kernel to be able to support this IPC version.
The idea is that when you have a product kernel then you can drop unused, dead code to make the kernel smaller.

If any other platform than Intel adds support to IPC4 then they can select it.

We can as well compile everything all the time as we do right now, I just thought that why compile code when it will never going to be used.

If all agree, we can keep this patch or drop the compile time selection of supported IPC versions. The ipc4 support in core is around 2K loc...

Reply:
@ujfalusi  Sorry for my poor explanation. I just want to suggest to rename SND_SOC_SOF_INTEL_IPC4 as SND_SOC_SOF_IPC4 and I don't have other concerns about this patch. Thanks.

Reply:
@yaochunhung, Ah, sorry for the misunderstanding, I have used this name as per the in kernel enum for the versions"
```c
/* Definitions for multiple IPCs */
enum sof_ipc_type {
	SOF_IPC,
	SOF_INTEL_IPC4,
	SOF_IPC_TYPE_COUNT
};
```

Reply:
@ujfalusi  Got it. thanks.